File: server/entities/theme.ts
import {
  Column,
  CreateDateColumn,
  Entity,
  JoinColumn,
  ManyToOne,
  PrimaryColumn,
  UpdateDateColumn,
} from 'typeorm';
import Config from './config';

@Entity()
export default class Theme {
  @PrimaryColumn({ nullable: false })
  id!: string;

  @Column({ nullable: true })
  label: string;

  @Column('simple-json')
  accent!: { primary: string; secondary: string };

  @Column('simple-json')
  background!: { primary: string; secondary: string };

  @Column('simple-json')
  text!: { primary: string; secondary: string };

  @Column('simple-json')
  border!: { primary: string; secondary: string };

  @ManyToOne(() => Config)
  @JoinColumn({ referencedColumnName: 'id' })
  config?: Config;

  @CreateDateColumn()
  createdAt?: Date;

  @UpdateDateColumn()
  updatedAt?: Date;
}

File: server/pkg/theme/theme.go
package theme

import (
	"github.com/gin-gonic/gin"
	"gorm.io/datatypes"
	"gorm.io/gorm"
)

type Theme struct {
	ID         string         `json:"id"`
	Label      string         `json:"label"`
	Accent     datatypes.JSON `json:"accent"`
	Background datatypes.JSON `json:"background"`
	Text       datatypes.JSON `json:"text"`
	Border     datatypes.JSON `json:"border"`
}

func NewService(ctx *gin.Engine, db *gorm.DB) Theme {

	theme := Theme{}
	theme.Routes(ctx, db)

	return theme

}
